### ISSUE TITLE
Training on DEM SRTM dataset

### ISSUE THREAD

**Issue by htn274** [JiahuiYu/generative_inpainting#451]
Thank you for your great work. I have been applying your research to my problem now.
Here is my problem:

> DEM, stands for Digital Elevation Model, is a numerical matrix with each pixel represents its elevation correspond to a location.

SRTM is a globally DEM dataset. However, with a reservoir or a lake, SRTM showed the elevation of surface water at the time it collected (in 2000). Now I want to recover DEM below surface water of a lake or a reservoir.

Here is my step-by-step process:

1. My training set is 15k DEM images.

2. I generated masks correspond to input image. **An input image has a specific mask**. My mask is an connected area which has elevation lower than a random number. So it can describe a reservoir or a lake, I think so. Some of my masks with the input:

![image](https://user-images.githubusercontent.com/26211514/85963678-0ae85580-b9e1-11ea-8b9c-3e6c44c9be60.png)

So I customized your code to read my input: 
![image](https://user-images.githubusercontent.com/26211514/85963740-47b44c80-b9e1-11ea-8e90-aecdfbc8fe99.png)

![image](https://user-images.githubusercontent.com/26211514/85963808-89dd8e00-b9e1-11ea-9bc0-77572b178022.png)

3. I also customized **data_from_fnames** in **neuralgym** toolkit to read '**.tif'** files. Then I used min-max normalize DEM image to 0-255 in order to pass it to your model.

4. Some first rows in my train.flist:

![image](https://user-images.githubusercontent.com/26211514/85964135-ae863580-b9e2-11ea-9d76-289d95533a41.png)

5. Because of limit memory of my GPU, I set image input shape (128, 128, 1) and batch size is 16 as your advice I saw in other issues. 

```
#training
train_spe: 1000 
max_iters: 1000000
viz_max_out: 10
val_psteps: 500

static_view_size: 30
img_shapes: [128, 128, 1]
height: 128
width: 128
max_delta_height: 32
max_delta_width: 32
batch_size: 16
vertical_margin: 0
horizontal_margin: 0
```

6. I finally trained more than 70 epoches and get bad results:

My losses are not converging

![image](https://user-images.githubusercontent.com/26211514/85965302-797be200-b9e6-11ea-9cfc-83f30fff766d.png)

![image](https://user-images.githubusercontent.com/26211514/85965349-96b0b080-b9e6-11ea-8dbe-a78108af6ca3.png)

Some of generated validation images:

![image](https://user-images.githubusercontent.com/26211514/85966457-0d02e200-b9ea-11ea-8cc9-933b66cec186.png)

**Questions**

1. Are all my customizations correct? 
2. I saw your sample flist is shuffled and I didn't. Does it affect to the training result?
3. Could you give me some suggestion and your views on my problem?

Thanks a lot! <3

**Comment by Efrat-Taig** [comment]
Hi htn274
This seems like a nice idea, how did it work out for you?
I am trying to implement a similar idea, with non random masks, and  it can be very helpful to learn from your experience.

Thanks in advance

**Comment by JiahuiYu** [comment]
@htn274 Thanks for your detailed feedback and that's exactly our community needs! I will keep this issue in the front page.

Regarding your question:
1. I think you are right, and the results are good already?
2. Shuffling should always be non-worse.
3. Larger dataset could help.

**Comment by Nico-Adamo** [comment]
Hi there, I'm wondering exactly how you modified the inpaint network to work with this - as I currently see it, the network only works with a mask of size (1, height, width,1), but with your custom masks you're feeding it (batch_size, height, width, 1). I'm trying to do the  same thing  but getting a "Dimensions must be equal" ValueError. Any guidance?

**Comment by Nico-Adamo** [comment]
Ah nevermind, I figured it out! For future reference, there's a lot more modifications than those shown to get custom masks working - but it's not too difficult. My code is a bit crazy right now but if anyone needs an explanation, feel free to ping me.

**Comment by Efrat-Taig** [comment]
Hey! I will be happy  to hear your explanation. How can I contact you?
Efrat (efrat@bria.ai)

**Comment by SvSz** [comment]
How Exciting! This is almost exactly what I am trying to do!

@htn274  @Nico-Adamo @Efrat-Taig I am eager to hear how you did and exchange some results and other experiences!
you can PM me via twitter e.g.

**Comment by Baumflaum** [comment]
As described in #444 you will have to change the batch size to 1 since one mask always services the whole batch. Also you will have to change the call in `build_graph_with_losses` to:

```
  x1, x2, offset_flow = self.build_inpaint_net(
      xin, mask, reuse=tf.AUTO_REUSE, training=training,
      padding=FLAGS.padding)
```
